apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -1,116 +1,74 @@
 import GoogleBlockly from 'blockly/core';
 
-const CORNER_RADIUS = 3;
-const INNER_HEIGHT = 16;
-
+/**
+ * This is a customized field which the user clicks to select an option from a customized picker,
+ * for example, the location of a sprite from a grid or a sound file from a customized modal.
+ * @param value Optional. The initial value of the field.
+ * @param validator Optional. A function that is called to validate changes to the field's value.
+ * Takes in a value & returns a validated value, or null to abort a change
+ * @param onClick The function that handles the field's editor.
+ * @param transformText Optional. The function that handles how the field text is displayed.
+ * @param icon Optional. SVG <tspan> element - if the field displays a button, this is the icon that is displayed on the button.
+ */
 export default class CdoFieldButton extends GoogleBlockly.Field {
-  // Third argument (color) is used by CDO blockly, so we include it in
-  // method signature for compatibility
-  constructor(title, opt_buttonHandler, _, opt_changeHandler) {
-    super('');
-
-    this.title_ = title;
-    this.buttonHandler_ = opt_buttonHandler;
-    this.changeHandler_ = opt_changeHandler;
+  constructor(options) {
```

Comment:
nit: you can destruct the options param inline for readability if that's clearer, like 
```
constructor({value, validator, onClick ... }) {
  this.onClick = onClick;
  ...
}

```

Reply:
Interesting! Will do.